### ISSUE TITLE
First file error message

### ISSUE THREAD

**Issue by TacticalLettuce** [FrexG/EEG-Seizure-Detection#1]
If I am trying to save the the `bipolar_montage` value as an `.npz` file with the necessary annotations, it gets this error:

`Traceback (most recent call last):
  File "c:\Users\mario\AppData\Local\Programs\Python\Python310\lib\site-packages\joblib\externals\loky\process_executor.py", line 428, in _process_worker
    r = call_item()
  File "c:\Users\mario\AppData\Local\Programs\Python\Python310\lib\site-packages\joblib\externals\loky\process_executor.py", line 275, in __call__
    return self.fn(*self.args, **self.kwargs)
  File "c:\Users\mario\AppData\Local\Programs\Python\Python310\lib\site-packages\joblib\_parallel_backends.py", line 620, in __call__
    return self.func(*args, **kwargs)
  File "c:\Users\mario\AppData\Local\Programs\Python\Python310\lib\site-packages\joblib\parallel.py", line 288, in __call__
    return [func(*args, **kwargs)
  File "c:\Users\mario\AppData\Local\Programs\Python\Python310\lib\site-packages\joblib\parallel.py", line 288, in 
    return [func(*args, **kwargs)
  File "C:\Users\mario\AppData\Local\Temp\ipykernel_22432\857830057.py", line 35, in compress_and_save
  File "<__array_function__ internals>", line 180, in savez_compressed
  File "c:\Users\mario\AppData\Local\Programs\Python\Python310\lib\site-packages\numpy\lib\npyio.py", line 666, in savez_compressed
    _savez(file, args, kwds, True)
  File "c:\Users\mario\AppData\Local\Programs\Python\Python310\lib\site-packages\numpy\lib\npyio.py", line 696, in _savez
    val = np.asanyarray(val)
  File "c:\Users\mario\AppData\Local\Programs\Python\Python310\lib\site-packages\mne\io\base.py", line 834, in __getitem__
    return self._getitem(item)
  File "c:\Users\mario\AppData\Local\Programs\Python\Python310\lib\site-packages\mne\io\base.py", line 837, in _getitem
    sel, start, stop = self._parse_get_set_params(item)
  File "c:\Users\mario\AppData\Local\Programs\Python\Python310\lib\site-packages\mne\io\base.py", line 770, in _parse_get_set_params
    sel = _picks_to_idx(self.info, item[0])
  File "c:\Users\mario\AppData\Local\Programs\Python\Python310\lib\site-packages\mne\io\pick.py", line 1350, in _picks_to_idx
    raise ValueError(
ValueError: All picks must be < n_channels (22), got 22
"""`

**Comment by FrexG** [comment]
Were you running it on the notebook?, because i couldn't reproduce your error!

**Comment by FrexG** [comment]
Were you running it on the notebook?, because i couldn't reproduce your error!

**Comment by TacticalLettuce** [comment]
Yeah, I was running on a Jupiter notebook in visual studio code. Basically I attempt to save the bipolar array as an mne object, and the function throws the error, but if I instead save the data part of the bipolar array as an numy array, it runs without errors. The error is in np.savez_compressed(), so maybe is some incompatibility with the mne arrays on numpy, or I have different versions, but the notebook is the same as the uploaded on the repository. I am also getting some errors on the second notebook when I solve the first problem. Any idea?

**Comment by FrexG** [comment]
Ohh, np.saves_compressed() can only be used with numpy arrays that's why. Can you tell where you get the error when running the second notebook(i am assuming it is `2.0-annotation-eda.ipynb`).

**Comment by TacticalLettuce** [comment]
Well, I am traying to create the window in the cell that is below the text _Filter reading with only `bckg` annotation for the whole sample duration._, and it seems thay it works fine for the files named `dev_channel_based_annotation.json` and `train_channel_based_annotation.json`, but for the same files generated in the previous notebook, it gets several errors as it doesn't read the format correctly. I am trying to fix it but it isn't something that works fine, at least in my case

**Comment by TacticalLettuce** [comment]
Actually, this error is the root cause of the mayority of errors on 2.0-annotation-eda and the following notebooks. It seems as if you have mixed different commits of the notebooks, because the formats of the files that the first notebook creates are totally uncompatible with the format that the following notebooks require. Also, the creation of the channel_based annotations doesn't work propertly, because when a channel have more than 1 event, it creates a sublist of those events in the channel, but the other channels are tagged with one event, so all the channels of one recording have different shapes.  
Is my explanation understandable? Sorry if I'm bothering with my inquieres. I would like to help as this repo is really useful to me

**Comment by FrexG** [comment]
I really love your engagement on this repo since i am also just started working on EEG data and didn't think anyone will find it. That's why it a bit messy. I tried fixing the issue with notebook, you were correct they were mix up of commits. I am not sure if i understand your question on the the channe_based annotation but you can make the channels of recording to have the same shape by using a sliding window. I selected a window size of $t = 1$ second, and create a non-overlapping sample of length of $t \times s_{freq}$. You can then create a `y` label for each window by referring to the annotation file for each channel.

**Comment by TacticalLettuce** [comment]
Hi again. I have been doing some changes on the repo to make the code work, and I have partially made it. I was able to make filtered annotations, but when I had to run the XGBoost model, I ran out of memory due to the model exceed my available ram. I tried to limit the values of the x_train_np but it didn't seem to work. So maybe there is missing some code or is not as easy as I thought. However, my issue was solved, so I can close it. If you came across with a solution, I will be happy to try it again and see if it works!
